stdlib/public/core/StringInterpolation.swift

```diff
@@ -197,6 +197,46 @@ public struct DefaultStringInterpolation: StringInterpolationProtocol, Sendable
   }
 }
 
+extension DefaultStringInterpolation {
+  @backDeployed(before: SwiftStdlib 6.2)
+  public mutating func appendInterpolation<T>(
+    _ value: T?, 
+    default: @autoclosure () -> String
+  ) where T: TextOutputStreamable {
+    if let value {
+      self.appendInterpolation(value)
+    } else {
+      self.appendInterpolation(`default`())
+    }
+  }
+
+  @backDeployed(before: SwiftStdlib 6.2)
+  @_disfavoredOverload
```

Comment:
Instead of using `@_disfavoredOverload` for these, you should be able to provide a fourth `T: TextOutputStreamable, T: CustomStringConvertible` overload. That's how regular `appendInterpolation(_:)` works. (The type checker *really* penalizes disfavored overloads, so it's usually better to avoid it if you can.)

Also, consider making these `@inlinable`. When I originally implemented `DefaultStringInterpolation`, I did some interpolation performance benchmarking (the test should still be in the benchmark suite) and found that inlining really matters for it.

Reply:
Oh, I didn't realize! I thought I was helping out by only providing three overloads instead of four. 🙃 I'll look at adding `@inlinable`, too. Thanks!